### ISSUE TITLE
Resized viewport sometimes not preserved after page navigation

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [DevExpress/testcafe#4145]
### What is your Test Scenario?
There is an element that is only visible when testing the mobile version of our website. We resize the browser window during tests, navigate to the page, then attempt to click on it.

Note that the tests _also_ have video recording enabled. 
 
### What is the Current behavior?
Sometimes, somewhat uncommonly -maybe 1 in 10-20 times?- the browser viewport will revert back to the original size after a page navigation. You can see in the video below that the page is mobile when first resized, then switches back to being full sized after the page navigation.
It's very spooky. 👻
 
### What is the Expected behavior?
We're aware of the notice that video recordings aren't fully supported from browser viewport resizing, but we need these recordings to help debug CI failures. Either the page should still work despite video weirdness _or_ there should be a way to specify viewport size on the command-line. Otherwise we have no way to _both_ test mobile viewports _and_ always have recordings, even for nondeterministic failures from flaky tests.
 
### What is your web application and your TestCafe test code?
I'm sorry, we're not open source yet 😢. But if you email me at josh@codecademy.com I can provide you with a .zip file of our repository to help debug.

Your website URL (or attach your complete example): https://codecademy.com

<details>
<summary>Here's roughly what the test looks like:</summary>

```js
await t.resizeWindowToFitDevice('iPhone 6 Plus');
await t.navigateTo('https://codecademy.com/about');
await t.click(getTestElement('header-mobile-menu'));
```
</details>

<details>
<summary>Your complete test report:</summary>

```
   1) The element that matches the specified selector is not visible.

      Browser: HeadlessChrome 76.0.3809 / Linux 0.0.0
      Screenshot:
      /tmp/testcafe-artifacts/test-7-4__HeadlessChrome/Terms-errors/1.png

         65 |test('has no accessibility complaints when loaded as an
      authenticated user on a mobile viewport and the profile photo is
      clicked', async t => {
         66 |  await useRegisteredUserRole(t);
         67 |  await t.navigateTo(testURL);
         68 |  await useMobileViewport(t);
         69 |
       > 70 |  await t.click(getTestElement('header-mobile-menu'));
         71 |
         72 |  await axeCheck(t);
         73 |});
         74 |

         at click

   (/root/project/webpack/acceptanceTests/landingPages/terms.spec.js:70:11)
         at step

   (/root/project/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
         at <anonymous>
      (/root/project/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13)



 1/7 failed (1m 52s)

 Warnings (7):
 --
  The browser window was resized during the "has no accessibility complaints
  when loaded as an anonymous user on a desktop viewport" test while TestCafe
  recorded a video. TestCafe cannot adjust the video resolution during
  recording. As a result, the video content may appear broken. Do not resize
  the browser window when TestCafe records a video.
 --
  ...(more of the same - there are 7 similar tests, and only one is failing)...

```
</details>

<details>
<summary>Screenshots:</summary>

GIF version of the recorded video, since GitHub doesn't allow embedding videos:
![ezgif-1-2d1e29bc40e0](https://user-images.githubusercontent.com/3335181/62945169-3b64d800-bdac-11e9-92b5-9848cd6fc773.gif)
See [Terms-1.zip](https://github.com/DevExpress/testcafe/files/3497099/Terms-1.zip) for the actual recording.

TestCafe artifact from time of failure:

![1](https://user-images.githubusercontent.com/3335181/62944782-68fd5180-bdab-11e9-8b27-e14015cb1c71.png)

</details>

### Steps to Reproduce:

1. Create roughly the above test
2. Run it in CircleCI a couple dozen times with test recording enabled
3. Notice failures 1-2 times
4. 😭
 
### Your Environment details:
 
* testcafe version:                   1.3.3
* node.js version:                    8.10.0
* command-line arguments:  NODE_ENV=development $(yarn bin)/testcafe "webpack/acceptanceTests/**/*.spec.js"
* browser name and version: HeadlessChrome 76.0.3809
* platform and version:          Linux Ubuntu 14.04
* other:                                   Running in CircleCI

**Comment by devmondo** [comment]
yes, we noticed this behavior too, by is there a way to maximize the browser before starting any test without resorting to `t.maximize`, like what @JoshuaKGoldberg said? 

@JoshuaKGoldberg i laughed so hard at the ghost icon you added to the below
It's very spooky. 👻

**Comment by helen-dikareva** [comment]
@JoshuaKGoldberg Thank you for such a detailed description of the issue.
We didn't face such behavior. It can be related to some bug in Chrome. Could you specify if this problem can be reproduced locally or only in CI. And please provide us with your working project and test code to reproduce the issue. You can send it at support@devexpress.com.

We'll consider implementation of the options for viewport size adjustment for all tests globally. But now we think that it could be a case for using [a global setup action](https://github.com/DevExpress/testcafe/issues/745).
Also please note that you can use the [device emulation](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/browsers/using-chrome-device-emulation.html) for testing only in Chrome.

The warnings that you can see in a test report only warn you that if you use resize actions in tests during video recording, you'll get a wrong aspect ratio for the resulting video.

**Comment by JoshuaKGoldberg** [comment]
> Could you specify if this problem can be reproduced locally or only in CI

This only happens in CI. We run tests on local machines -Mac, Linux, and Windows- and in CircleCI -Ubuntu Linux-; it's only shown up in CircleCI.

**Comment by helen-dikareva** [comment]
Thank you for this clarification.

**Comment by helen-dikareva** [comment]
@JoshuaKGoldberg we received your project, thank you for your cooperation. We need some time to investigate it. We'll concact you if we need any clarifications and we notify you as soon as we get any results.

**Comment by Farfurix** [comment]
@JoshuaKGoldberg 

Hello,
 
I tried to run your tests locally and they failed with some errors.
Then I fixed your tests:
1. Wrong imports:
```diff
- import axeCheck from 'axe-testcafe';
+ import { axeCheck } from 'axe-testcafe';
```
2. Wrong paths:
```diff
- import { config, endpoints } from '../config';
+ import { config, endpoints } from '../../config';
```
3. Install all required dependencies.

The "local run" result had a lot of failed assertions.

Could you please send us (support@devexpress.com) your **complete project**, not only the tests. If I understand you correctly, the complete project will allow me to see all the passed tests in the case of the "local run" and failed tests in the case of "Circle CI run".

The complete project should have:
1. The list of all dependencies
2. The config files with the required TestCafe commands to reproduce the issue on Circle CI
3. The Circle CI config file
4. Only required tests

**Comment by JoshuaKGoldberg** [comment]
I emailed the full repro case this week. Could someone reopen this please?

**Comment by alexey-lin** [comment]
Hi @JoshuaKGoldberg, we've received your e-mail, thank you. We need some time to investigate the project. We'll let you know if we need some additional information.

**Comment by Farfurix** [comment]
@JoshuaKGoldberg 

In your latest project, I cannot see the following:
1. The list of all dependencies
2. The config files with the required TestCafe commands to reproduce the issue on Circle CI
3. The Circle CI config file
4. Are there only required tests?

Please provide us with a complete project that I described here: https://github.com/DevExpress/testcafe/issues/4145#issuecomment-523833045.

**Comment by JoshuaKGoldberg** [comment]
EmaileD!

**Comment by Farfurix** [comment]
@JoshuaKGoldberg 

Hello,
 
We would love to assist you and diagnose this issue. However, our policy prevents us from accessing internal resources without prior written approval from a site owner. If you want us to further research the problem directly on your side, please ask the website owner to send us (mailto:support@devexpress.com) a written confirmation. It must permit DevExpress personnel to remotely access the website and its internal resources for research/testing/and debugging purposes.

**Comment by JoshuaKGoldberg** [comment]
@Farfurix I've already done that over email as a site owner. I'll follow up on the email thread to see if you need more.

Also, could you make these asks over email? It feels like we're spamming this legitimate issue thread with user-specific queries.

**Comment by Farfurix** [comment]
@JoshuaKGoldberg 

Thank you, we received the confirmation.
Our team needs some time to examine your project. Please stay tuned for updates in this thread.

**Comment by JoshuaKGoldberg** [comment]
It's been almost 2 years and I no longer use Testcafe. Closing to keep my issue queue small. Cheers! 👋
